File: codewars/6kyu/ticker/dduck-ticker.js
// https://www.codewars.com/kata/ticker/train/javascript

const getTick = (text, width, tick, count = tick) => {
  if (count <= 0) {
    return text.slice(0, width);
  }
  return getTick(text.slice(1) + text[0], width, tick, count - 1);
};

const dduckTicker = (text, width, tick) => {
  text = (''.padStart(width) + text);
  const quotient = Math.floor((tick || 0) / text.length);
  return getTick(text, width, quotient === 0 ? tick : tick % text.length);
};

console.log(dduckTicker('Beautiful is better than ugly.', 10, 0), '          ');
console.log(dduckTicker('Beautiful is better than ugly.', 10, 5), '     Beaut');
console.log(dduckTicker('Beautiful is better than ugly.', 10, 30), 'than ugly.');
console.log(dduckTicker('Beautiful is better than ugly.', 10, 39), '.         ');
console.log(dduckTicker('Beautiful is better than ugly.', 10, 41), '         B');
console.log(dduckTicker('under off part! home! us', 18, 26219));

File: codewars/6kyu/ticker/dduck-ticker.py
# https://www.codewars.com/kata/ticker/train/javascript

import unittest


def get_ticker(text, width, tick, count):
    if count == 0:
        return text[0:width]

    return get_ticker(text[1:] + text[0], width, tick, count - 1)


def ticker(text, width, tick):
    text = "".rjust(width) + text
    quotient = tick // len(text)
    remainder = tick % len(text)
    target_tick = tick if quotient == 0 else remainder

    return get_ticker(text, width, target_tick, target_tick)


unittest.TestCase().assertEqual(ticker("Beautiful is better than ugly.", 10, 0), "          ")
unittest.TestCase().assertEqual(ticker("Beautiful is better than ugly.", 10, 5), "     Beaut")
unittest.TestCase().assertEqual(ticker("Beautiful is better than ugly.", 10, 30), "than ugly.")
unittest.TestCase().assertEqual(ticker("Beautiful is better than ugly.", 10, 39), ".         ")
unittest.TestCase().assertEqual(ticker("Beautiful is better than ugly.", 10, 41), "         B")

unittest.TestCase().assertEqual("test-string", "test-some-string")

"""
beat practice in codewars

def ticker(text, width, tick):
    tick %= len(text) + width
    text = ' ' * width + text + ' ' * width
    return text[tick :  tick + width]
"""
